app/src/main/java/com/example/campusguide/DirectionsActivity.kt

```diff
@@ -76,21 +97,21 @@ class DirectionsActivity : AppCompatActivity() {
             when (view.id) {
                 R.id.radio_driving ->
                     if (checked) {
-                        travelMode = "Driving"
-                        path = createPath(startName, endName, travelMode)
-                        setPathOnMapAsync(path)
+                        removePreviousPath()
```

Comment:
you could save couple lines if you can extract those radio button lambdas to a method.

Reply:
It'll save us a couple lines but make the code less understandable imo
I'll leave it as is but keep your suggestion in mind if the lambdas get more complex